tests/test_edgeql_functions.py

```diff
@@ -3631,6 +3631,24 @@ async def test_edgeql_functions_string_bytes_conversion_error(self):
                 ''',
             )
 
+    async def test_edgeql_functions_json_bytes_conversion(self):
+        string = json.dumps({"a": [1, 2, 3], "b": "foo"})
+
+        await self.assert_query_result(
```

Comment:
Love a good roundtrip test case 👏 . Might be worth adding some weird stuff like utf-8 characters maybe?